text/0000-bundle-groups.md

```diff
@@ -0,0 +1,89 @@
+
+Alternatively, you can use the `--only` option (provided on both the `install` and `deploy` commands) to install only the groups whose names you give at that time. For example, if you were to run `bundle install --only production`, the `test` group would not be installed even though it is enabled by default, and the `production` group would be installed even though it is disabled by default.
+
+# Reference-level explanation
+
+The `groups` command can be backed by the same Bundler settings as always, with keys of WITH and WITHOUT. It could alternately be backed by separate settings keys for any group set to a different status than the default status declared in the Gemfile. Either way, I think it amounts to an implementation detail that users won’t care about.
+
+Switching to the `groups` command completely removes ambiguity about whether a group should be enabled or disabled… a group should be enabled if it was passed as an argument to `groups enable`, and disabled when it’s passed as an argument to `groups disable`.
+
+I also propose that this command would allow us to deprecate and remove the `install_if` feature. We could extend the `groups` command guide to explain how to use optional groups and a build script that runs a check and executes `bundle groups enable foo` before running `bundle install`. That removes one (maybe the only?) hard blocker for non-executable Gemfiles. /cc @segiddins
+
+# Drawbacks
+
+This is definitely a backwards-breaking change. We could keep the existing plan of having users set enabled and disabled groups via `bundle config with foo` and `bundle config without foo`.
+
+I think the downsides to this approach in general are that it will take a decent amount of work in the form of implementation, docs, and training users.
+
+# Rationale and Alternatives
+
+I think this approach is the clearest, least surprising, most easily explainable, and most flexible way of handling this user need. The `--with` option is basically a hack directly on top of the `--without` system that didn’t fully meet user needs, and the proposed `--only` option is an additional hack on top of those layered hacks trying to meet further user needs.
+
+We’ve considered other options throughout the history of Bundler, but those options have almost entirely boiled down to “keep the options we have that aren’t flexible enough” or “accept additional options that dramatically increase complexity”.
+
+The impact of not doing this is that we would be maintaining the status quo, which is… not great, for `with` and `without` groups.
+
+# Unresolved questions
+
+How should we store which groups are enabled and disabled? Should we keep the existing settings keys or use new keys?
+
+Should we include the `--only` option in the final implementation of this RFC, or skip it?
```

Comment:
Can be decided when you release an beta version of this feature and get more feedback. It's nice-to-have feature but not a show-stopper.